File: 27.remove-element.js
/*
 * @lc app=leetcode id=27 lang=javascript
 *
 * [27] Remove Element
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
  let i = 0;
  for (const num of nums) {
    if (num !== val) {
      nums[i++] = num;
    }
  }
  return i;
};
// @lc code=end

File: 27.remove-element.go
/*
 * @lc app=leetcode id=27 lang=golang
 *
 * [27] Remove Element
 */

package leetcode

// @lc code=start
func removeElement(nums []int, val int) int {
	i := 0
	for _, num := range nums {
		if num != val {
			nums[i], i = num, i+1
		}
	}
	return i
}

// @lc code=end
